crates/ruff_python_parser/tests/snapshots/invalid_syntax@expressions__dict__double_star.py.snap

```diff
@@ -0,0 +1,641 @@
+                                    ExprName {
+                                        range: 256..257,
+                                        id: "x",
+                                        ctx: Load,
+                                    },
+                                ),
+                                Name(
+                                    ExprName {
+                                        range: 265..266,
+                                        id: "y",
+                                        ctx: Load,
+                                    },
+                                ),
+                            ],
+                        },
+                    ),
+                },
+            ),
+            Expr(
+                StmtExpr {
+                    range: 268..277,
+                    value: Dict(
+                        ExprDict {
+                            range: 268..277,
+                            keys: [
+                                None,
+                                Some(
+                                    Name(
+                                        ExprName {
+                                            range: 275..276,
+                                            id: "y",
+                                            ctx: Load,
+                                        },
+                                    ),
+                                ),
+                            ],
+                            values: [
+                                Name(
+                                    ExprName {
+                                        range: 271..272,
+                                        id: "x",
+                                        ctx: Load,
+                                    },
+                                ),
+                                Name(
+                                    ExprName {
+                                        range: 276..276,
+                                        id: "",
+                                        ctx: Invalid,
+                                    },
+                                ),
+                            ],
+                        },
+                    ),
+                },
+            ),
+        ],
+    },
+)
+```
+## Errors
+
+  |
+2 | # the ones which are higher than that.
+3 | 
+4 | {**x := 1}
+  |      ^^ Syntax Error: expected Comma, found ColonEqual
+5 | {a: 1, **x if True else y}
+6 | {**lambda x: x, b: 2}
+  |
+
+
+  |
+2 | # the ones which are higher than that.
+3 | 
+4 | {**x := 1}
+  |          ^ Syntax Error: expected Colon, found Rbrace
+5 | {a: 1, **x if True else y}
+6 | {**lambda x: x, b: 2}
+  |
+
+
+  |
+4 | {**x := 1}
+5 | {a: 1, **x if True else y}
+  |            ^^ Syntax Error: expected Comma, found If
+6 | {**lambda x: x, b: 2}
+7 | {a: 1, **x or y}
+  |
+
+
+  |
+4 | {**x := 1}
+5 | {a: 1, **x if True else y}
+  |                          ^ Syntax Error: Expected a statement
```

Comment:
I'm a bit surprised by the recovery here. Should it not recognize that `if` is a valid expression start and parse the expression instead?

Reply:
The minimum precedence for an expression after `**` in a dictionary is bitwise or (`|`) and `if` expressions are lower than that. The parser won't parse the `if ...` because the grammar doesn't allow it. This is one of the areas which can possibly be changed when working on error recovery.